MotionPhotoReader/app/src/main/java/com/google/android/libraries/motionphotoreader/MotionPhotoInfo.java

```diff
@@ -0,0 +1,129 @@
+package com.google.android.libraries.motionphotoreader;
+
+import android.media.MediaExtractor;
+import android.media.MediaFormat;
+import android.os.Build;
+
+import androidx.annotation.RequiresApi;
+import androidx.annotation.VisibleForTesting;
+
+import com.adobe.internal.xmp.XMPException;
+import com.adobe.internal.xmp.XMPMeta;
+
+import java.io.File;
+import java.io.FileDescriptor;
+import java.io.FileInputStream;
+import java.io.IOException;
+
+/**
+ * Holds data about the Motion Photo file.
+ */
+public class MotionPhotoInfo {
+
+    private final int width;
+    private final int height;
+    private final long duration;
+    private final int rotation;
+
+    private final int videoOffset;
+    private final long presentationTimestampUs;
+
+    private MediaExtractor extractor;
+
+    /**
+     * Creates a MotionPhotoInfo object associated with a given file.
+     */
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    private MotionPhotoInfo(MediaFormat mediaFormat, int videoOffset, long presentationTimestampUs) {
+        width = mediaFormat.getInteger(MediaFormat.KEY_WIDTH);
+        height = mediaFormat.getInteger(MediaFormat.KEY_HEIGHT);
+        duration = mediaFormat.getLong(MediaFormat.KEY_DURATION);
```

Comment:
this assignment can be simplified with a ternary operator:
```
rotation = mediaFormat.containsKey(MediaFormat.KEY_ROTATION) 
    ? mediaFormat.getInteger(MediaFormat.KEY_ROTATION)
    : 0;
```

Reply:
break lines within 100-character limit and also start new lines with operators (don't end with an operator.. google java style)